invenio_base/utils.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2017 CERN.
+#
+# Invenio is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+from flask import current_app
+from .signals import secret_key_changed
+from invenio_db import db
+from sqlalchemy import update
+from sqlalchemy_utils.types.encrypted import EncryptedType
+from sqlalchemy.engine import reflection
+
+
+def rebuild_encrypted_field(old_key, table, field):
```

Comment:
Btw you can just replace the `app.secret_key` and you don't need to worry about low-level db access.

```python
from invenio_oauth2server.models import *
app.secret_key = old_key
x = db.session.query(Token.id, Token.access_token).all()
db.session.expunge_all()
app.secret_key = app.config['SECRET_KEY']
for id_, value in x:
    Token.query.filter_by(id=id_).update({'access_token': value})
db.session.commit()
```